docs/development/index.rst

```diff
@@ -416,3 +415,30 @@ A lot of code that's not gamma-ray specific belongs in other packages
 (e.g. `Scipy`_, `Astropy`_, other Astropy-affiliated packages, `Sherpa`_).
 We currently have quite a bit of code that should be moved "upstream" or already has been,
 but the Gammapy code hasn't been adapted yet.
+
+Assert convention
+-----------------
+
+When performing tests, the preferred numerical assert method is
+`numpy.testing.assert_allclose`. Use
+
+``from numpy.testing import assert_allclose``
+
+at the top of the file and then just use ``assert_allclose`` for
+the tests. This makes the lines shorter, i.e. there is more space
+for the arguments.
+
+``assert_allclose`` covers all use cases for numerical asserts, so
+it should be used consistently everywhere instead of using the
+dozens of other available asserts from pytest or numpy in various
+places.
+
+More details in `numpy.testing.assert_allclose`.
+
+Tip: in case of assertion errors on arrays of quantity objects, such
+as `astropy.units.Quantity` or `~astropy.coordinates.Angle`,
+`assert_allclose` will not show the values. For debugging you can
+use the `.value` methods of these object types. Don't forget to
+remove the call to `.value` after performing the test.
+
+* TODO: should we submit a bug report to Astropy?
```

Comment:
Ok I filed the issue: https://github.com/astropy/astropy/issues/3978
and replaced the `Tip` and `TODO` by the reference. I didn't add an example, tough I added an equivalent method for `Angle`.